var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -38,78 +38,54 @@ class Mercury(CMakePackage):
     depends_on('bmi', when='+bmi')
     depends_on('mpi', when='+mpi')
     depends_on('libfabric@1.5:', when='+ofi')
-    depends_on('openpa@1.0.3:', when='+opa')
-    depends_on('boost@1.48:', when='+boost')
+    depends_on('openpa@1.0.3:', when='%gcc@:4.8')
+    depends_on('boost@1.48:', when='+boostsys')
+    depends_on('boost', when='@:0.9')  # internal boost headers were added in 1.0.0
+
+    conflicts('+ofi', when='@:0.9')    # libfabric support was added in 1.0.0
+    conflicts('~ofi', when='+udreg')   # udreg option is specific to OFI
 
     def cmake_args(self):
+        """Populate cmake arguments for Mercury."""
         spec = self.spec
-        args = [
-            '-DBUILD_SHARED_LIBS=ON',
+        variant_bool = lambda feature: str(feature in spec)
+        parallel_tests = '+mpi' in spec and self.run_tests
+
+        cmake_args = [
+            '-DBUILD_SHARED_LIBS=%s' % variant_bool('+shared'),
+            '-DBUILD_TESTING=%s' % str(self.run_tests),
+            '-DMERCURY_ENABLE_PARALLEL_TESTING=%s' % str(parallel_tests),
+            '-DMERCURY_USE_BOOST_PP=ON',
             '-DMERCURY_USE_CHECKSUMS=ON',
             '-DMERCURY_USE_EAGER_BULK=ON',
+            '-DMERCURY_USE_SELF_FORWARD=%s' % variant_bool('+selfforward'),
             '-DMERCURY_USE_SYSTEM_MCHECKSUM=OFF',
-            '-DMERCURY_USE_XDR=OFF'
+            '-DMERCURY_USE_XDR=OFF',
+            '-DNA_USE_BMI=%s' % variant_bool('+bmi'),
+            '-DNA_USE_CCI=%s' % variant_bool('+cci'),
+            '-DNA_USE_MPI=%s' % variant_bool('+mpi'),
+            '-DNA_USE_SM=%s'  % variant_bool('+sm'),
         ]
 
-        if '+boost' in spec:
-            args.append('-DMERCURY_USE_BOOST_PP=ON')
-        else:
-            args.append('-DMERCURY_USE_BOOST_PP=OFF')
-
-        if '+bmi' in spec:
-            args.append('-DNA_USE_BMI=ON')
-        else:
-            args.append('-DNA_USE_BMI=OFF')
-
-        if '+cci' in spec:
-            args.append('-DNA_USE_CCI=ON')
-        else:
-            args.append('-DNA_USE_CCI=OFF')
-
-        if '+mpi' in spec:
-            args.append('-DNA_USE_MPI=ON')
-        else:
-            args.append('-DNA_USE_MPI=OFF')
+        if '@1.0.0:' in spec:
+            cmake_args.extend([
+                '-DMERCURY_USE_SYSTEM_BOOST=%s'
+                % variant_bool('+boostsys'),
+                '-DNA_USE_OFI=%s' % variant_bool('+ofi'),
+            ])
 
         if '+ofi' in spec:
-            args.append('-DNA_USE_OFI=ON')
+            cmake_args.extend([
+                '-DNA_OFI_GNI_USE_UDREG=%s' % variant_bool('+udreg'),
```

Comment:
I prefer to use `append()` instead of `extend([])` if you're only adding a single element to a list.

Reply:
sure I will make that change.